src/cpp/fastdds/xtypes/type_representation/dds_xtypes_typeobjectCdrAux.ipp

```diff
@@ -196,6 +196,10 @@ eProsima_user_DllExport void deserialize(
                         {
                                                         case EK_COMPLETE:
                                                         case EK_MINIMAL:
+                                                            if (mid != 1)
```

Comment:
This is an issue fixed currently in fastddsgen master branch. We should remove it from this code.